clientconn.go

```diff
@@ -129,6 +132,13 @@ func WithBalancer(b Balancer) DialOption {
 	}
 }
 
+// WithServiceConfig returns a DialOption which has a channel to read the service configuration.
+func WithServiceConfig(c <-chan ServiceConfig) DialOption {
```

Comment:
So I think you are talking about the related interprocess communications for service config. That is not the thing here -- the writer of this channel is inside the same process. In this way, we avoid the JSON dependency in grpc package. We always try to minimize the dependency of grpc (credentials are in another separate package in grpc-go). I will send u a cl for this JSON thing and other related stuffs.